### PR TITLE
Slideshow - Add "SlideChanged" event

### PR DESCRIPTION
**Pull request by Walthzer**
**When merged this pull request will:**
- add an event upon the changing of a slide in a slideshow.

### CODE REVIEW

File: addons/slideshow/functions/fnc_autoTransition.sqf [hunk 1]
```diff
@@ -18,7 +18,7 @@
  * Public: No
  */
 
-params ["_objects", "_images", "_varString", "_duration"];
+params ["_objects", "_images", "_varString", "_currentSlideshow", "_duration"];
```
Comment (jokoho48) [inline addons/slideshow/functions/fnc_autoTransition.sqf:21]: header needs to be updated with new parameters

Comment (Walthzer) [reply addons/slideshow/functions/fnc_autoTransition.sqf:22]: Function header updated: 38d4c90eef4aa2b16d284318217ae0186eb943f3

File: addons/slideshow/functions/fnc_autoTransition.sqf [hunk 2]
```diff
@@ -34,6 +35,9 @@ missionNamespace setVariable [_varString, _currentSlide];
     _x setObjectTextureGlobal [0, _images select _currentSlide];
 } count _objects;
 
+//Raise event upon slide change
+[QGVAR(slideChanged), [_images select _currentSlide, _currentSlideshow]] call CBA_fnc_localEvent;
```
Comment (jonpas) [inline addons/slideshow/functions/fnc_autoTransition.sqf:39]: `_images select _currentSlide` save this into a variable on L33 and use it here and in a loop.

File: addons/slideshow/functions/fnc_createSlideshow.sqf [hunk 3]
```diff
@@ -100,5 +100,6 @@ if (_duration == 0) then {
     missionNamespace setVariable [_varString, 0];
 
     // Automatic transitions handler
-    [FUNC(autoTransition), [_objects, _images, _varString, _duration], _duration] call CBA_fnc_waitAndExecute;
+    [FUNC(autoTransition), [_objects, _images, _varString, _currentSlideshow, _duration], _duration] call CBA_fnc_waitAndExecute;
 };
+_currentSlideshow;
```
Comment (jonpas) [inline addons/slideshow/functions/fnc_createSlideshow.sqf:105]:
```suggestion

_currentSlideshow
```

File: addons/slideshow/functions/fnc_autoTransition.sqf [hunk 4]
```diff
@@ -34,6 +35,9 @@ missionNamespace setVariable [_varString, _currentSlide];
     _x setObjectTextureGlobal [0, _images select _currentSlide];
 } count _objects;
 
+//Raise event upon slide change
```
Comment (jonpas) [inline addons/slideshow/functions/fnc_autoTransition.sqf:38]:
```suggestion
// Raise event upon slide change
```
A bit of a redundant comment though.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/slideshow/functions/fnc_autoTransition.sqf (hunk 2 above)
Comment (Walthzer) [reply addons/slideshow/functions/fnc_autoTransition.sqf]: Added the variable add swapped all occurrences of  `_images select _currentSlide `95fb60a356789c4e179afdbbad07c19d94eac9d3

File: addons/slideshow/functions/fnc_autoTransition.sqf (hunk 4 above)
Comment (Walthzer) [reply addons/slideshow/functions/fnc_autoTransition.sqf]: Good point. removed it : 44498067e6a7654d944b4f4a52b75ebd91c97efb

**Review by PabstMirror — APPROVED** [review]

**Review by jonpas — APPROVED** [review]

**Pull request merged**